_source/_posts/2022-08-29-react-typescript-redux.md

```diff
@@ -0,0 +1,530 @@
+
+**Prerequisites**
+
+As of the time of this publication, Node >= 14.0.0 and npm >= 5.6 are required for Create React App. You can check for the latest required versions at [https://reactjs.org/docs/create-a-new-react-app.html](https://reactjs.org/docs/create-a-new-react-app.html).
+
+**NOTE:** Redux also recommends you install the [Chrome React DevTools Extension](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en).
+{% include toc.md %}
+## Create the React app using TypeScript
+
+We'll get started with the built-in React template for [Create React App](https://github.com/facebook/create-react-app) that uses Redux and Typescript. This includes the recommended Redux Toolkit and Redux's integration with React components.
+
+1. First, run:
+```bash
+npx create-react-app okta-react-redux --template redux-typescript
+```
+
+**NOTE:** Our demo repository uses React `^18.2.0` and React Scripts `5.0.1`.
+
+2. Then we'll add the Redux core by running:
+```bash
+npm add redux@4.2
+```
+
+3. Redux provides its own types, but we'll want to add our `react-redux` types since we're using Typescript:
+```bash
+npm add -D @types/react-redux@7.1
+```
+
+4. We'll also need routing (and routing types) for our app:
+```bash
+npm install react-router-dom@5
+npm add -D @types/react-router-dom@5.3
+```
+
+5. To start your app, run:
+```bash
+npm start
+```
+
+## How Redux works
+
+To update state using Redux, an action is dispatched and the store then uses the root reducer to calculate a new state as compared to the old state, which then notifies the proper subscribers of the update so that the UI can properly be updated.
+
+After the UI for your application is initially rendered using an initial global Redux store created using the root reducer, a typical redux workflow goes like this:
+
+1. An event in the UI such as a user button interaction happens in your app which triggers an action.
+2. The action is then dispatched to the store and an event handler that handles the needed logic makes the changes to your global state.
+3. The store notifies subscribed parts in the UI that there has been an update.
+4. Those subscribed parts then check to see if a re-render is needed to update whatever needs updating in the UI.
+
+**NOTE:** The built-in Redux template we used includes a great counter example of how Redux works. You can take a closer look at [Redux's explanation of the counter example](https://redux.js.org/introduction/examples/).
+
+### Useful terms to know when working with Redux
+
+**[Action](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#reducers)**
+
+An Action is an JS object with a type and a payload. It describes an actionable interaction within your application, and typically is named something like `createUser` or `addToDo`.
```

Comment:
```suggestion
An **action** is a JS object with a type and a payload. It describes an actionable interaction within your app, and typically is named something like `createUser` or `addToDo`.
```
Since the docs don't uppercase "action" i'm doing what they do instead - bolding.